src/adapter/src/coord/message_handler.rs

```diff
@@ -676,4 +697,105 @@ impl<S: Append + 'static> Coordinator<S> {
             });
         }
     }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    /// Finishes sequencing a command that was waiting on a real time recency timestamp.
+    async fn message_real_time_recency_timestamp(
+        &mut self,
+        conn_id: ConnectionId,
+        transient_revision: u64,
+        (compute_instance_id, compute_instance_name): (ComputeInstanceId, String),
+        id_bundle: CollectionIdBundle,
+        mut object_names: HashMap<GlobalId, String>,
+        real_time_recency_ts: mz_repr::Timestamp,
+    ) {
+        if let Some(real_time_recency_context) =
```

Comment:
The lightest of personal preference nits: do a manual early `return` if it's not present, instead of indenting the entire logic block.